best_practices/controllers.rst

```diff
@@ -115,6 +115,19 @@ for the homepage of our app:
         }
     }
 
+Fetching Services
+-----------------
+
+If you extend the base ``Controller`` class, you can access services directly from
+the container via ``$this->container->get()`` or ``$this->get()``. Instead, you
+should use dependency injection to fetch services: most easily done by
+:ref:`type-hinting action method arguments <controller-accessing-services>`:
+
+.. best-practice::
+
+    Don't use ``$this->get()`` or ``$this->container->get()`` to fetch services
```

Comment:
Here and in the previous paragraph we don't explain why you should do that. Should we add something or link to some other article in the docs?